pkg/client/cli/telepresence_test.go

```diff
@@ -859,6 +871,115 @@ func (is *interceptedSuite) TestB_ListingActiveIntercepts() {
+
+	// Verify that intercept remains but that no agent is found
+	assert.Eventually(func() bool {
+		stdout, _ := telepresence(is.T(), "--namespace", is.ns(), "list")
+		if match := rx.FindStringSubmatch(stdout); match != nil {
+			dlog.Infof(c, "Got match '%s'", match[1])
+			return strings.Contains(match[1], `No agent found for "hello-0"`)
+		}
+		return false
+	}, 5*time.Second, time.Second)
+
+	// Verify that volume mount is broken
+	_, err := os.Stat(filepath.Join(is.mountPoint, "var"))
+	assert.Error(err, "Stat on <mount point>/var succeeded although no agent was found")
+
+	// Scale up again (start intercepted pod)
+	assert.NoError(ts.kubectl(c, "--context", "default", "scale", "deploy", "hello-0", "--replicas", "1"))
+
+	// Verify that intercept becomes active
+	assert.Eventually(func() bool {
+		stdout, _ := telepresence(is.T(), "--namespace", is.ns(), "list")
+		if match := rx.FindStringSubmatch(stdout); match != nil {
+			return match[1] == "ACTIVE"
+		}
+		return false
+	}, 5*time.Second, time.Second)
+
+	// Verify that volume mount is restored
+	st, err := os.Stat(filepath.Join(is.mountPoint, "var"))
+	require.NoError(err, "Stat on <mount point>/var failed")
+	assert.True(st.IsDir(), "<mount point>/var is not a directory")
+}
+
+func (is *interceptedSuite) TestE_StopInterceptedPodOfMany() {
+	ts := is.tpSuite
+	assert := is.Assert()
+	require := is.Require()
+	c := dlog.NewTestContext(is.T(), false)
+	rx := regexp.MustCompile(fmt.Sprintf(`Intercept name\s*: hello-0-` + is.ns() + `\s+State\s*: ([^\n]+)\n`))
+
+	helloZeroPods := func() []string {
+		pods, err := ts.kubectlOut(c, "get", "pods", "--field-selector", "status.phase==Running", "-l", "app=hello-0", "-o", "jsonpath={.items[*].metadata.name}")
+		assert.NoError(err)
+		pods = strings.TrimSpace(pods)
+		dlog.Infof(c, "Pods = '%s'", pods)
+		return strings.Split(pods, " ")
+	}
+	// Get current pod
+	currentPods := helloZeroPods()
+	require.True(len(currentPods) == 1)
+	currentPod := currentPods[0]
+
+	// Scale up to two pods
+	require.NoError(ts.kubectl(c, "--context", "default", "scale", "deploy", "hello-0", "--replicas", "2"))
+	defer func() {
+		_ = ts.kubectl(c, "--context", "default", "scale", "deploy", "hello-0", "--replicas", "1")
+	}()
+
+	// Wait for second pod to arrive
+	assert.Eventually(func() bool { return len(helloZeroPods()) == 2 }, 5*time.Second, time.Second)
+
+	// Delete the currently intercepted pod
+	require.NoError(ts.kubectl(c, "--context", "default", "delete", "pod", currentPod))
+
+	// Wait for that pod to disappear
+	assert.Eventually(
+		func() bool {
+			for _, zp := range helloZeroPods() {
+				if zp == currentPod {
+					return false
+				}
+			}
+			return true
+		}, 5*time.Second, time.Second)
+
+	// Verify that intercept is still active
+	assert.Eventually(func() bool {
+		stdout, _ := telepresence(is.T(), "--namespace", is.ns(), "list", "--intercepts")
```

Comment:
One thing I'd add to this test is to curl the service to validate that you are getting the intercepted payload and not what's in the cluster (E.g. in case 2 of my comment on the PR, it says "ACTIVE" for the intercept but all requests are going to the service in the cluster).

Reply:
Added.